caikit/runtime/servicers/global_predict_servicer.py

```diff
@@ -265,14 +281,32 @@ def predict_model(
                 should be passed
             model (Optional[ModuleBase]):
                 Pre-fetched model object
+            request_id (Optional[str]):
+                A unique request ID for this request for tracing
             **kwargs: Keyword arguments to pass to the model's run function
         Returns:
             response (Union[DataBase, Iterable[DataBase]]):
                 The object (unary) or objects (output stream) produced by the
                 inference request
         """
+        trace.set_tracer(context, self._tracer)
+        trace_context = trace.get_trace_context(context)
+        trace_span_name = f"{__name__}.GlobalPredictServicer.predict_model"
+        with self._handle_predict_exceptions(
+            model_id, request_name
+        ), self._tracer.start_as_current_span(
+            trace_span_name,
+            context=trace_context,
+        ) as trace_span:
+
+            # Set trace attributes available before checking anything
+            trace_span.set_attribute("calling", trace_span_name)
+            trace_span.set_attribute("model_id", model_id)
+            trace_span.set_attribute("request_name", request_name)
+            trace_span.set_attribute("task", getattr(task, "__name__", str(task)))
+            if request_id:
```

Comment:
at this point, we would always have `request_id` right ?  Since each of functions, http server and grpc servicer are creating it ? So shall we make request_id as required field and remove this `if` condition ?